src/main/java/adris/altoclef/eventbus/events/ChatMessageEvent.java

```diff
@@ -1,14 +1,18 @@
 package adris.altoclef.eventbus.events;
 
 import net.minecraft.network.packet.s2c.play.ChatMessageS2CPacket;
+import java.lang.String;
 
 /**
  * Whenever chat appears
  */
 public class ChatMessageEvent {
     ChatMessageS2CPacket packet;
+    public String messageString;
+
 
     public ChatMessageEvent(ChatMessageS2CPacket packet) {
         this.packet = packet;
+        this.messageString = packet.body().content();
```

Comment:
```suggestion


    public ChatMessageEvent(ChatMessageS2CPacket packet) {
        this.packet = packet;
    
    public String messageContent {
        return packet.body().content();
    }
```

Reply:
i heard it's a good idea to follow the code style. on the older version, this is the code style

Reply:
> i heard it's a good idea to follow the code style. on the older version, this is the code style

Not at all, the current implementation contains a data abstraction violation: `ChatMessageEvent` represents the actual event, and the event does not contain a `messageString` attribute. Creating a method to return the value would be more conventional.

Reply:
but packet uses that way, why isn't packet also a method?

Reply:
> but packet uses that way, why isn't packet also a method?

Because `packet` is received when constructing the object.

Reply:
wait... why do we need messageString? why not get it on the event subscription on butler?

Reply:
> wait... why do we need messageString? why not get it on the event subscription on butler?

I don't know, your implementation... But if I were to guess, abstraction is probably the answer.

Reply:
let's leave it as is. it works fine and is same as the old code but String instead of Text

Reply:
> let's leave it as is. it works fine and is same as the old code but String instead of Text

Just because something works, doesn't mean it is good code. In 1.18.2, the text was passed when we are creating an object. But in newer versions, this doesn't happen.

Reply:
shall we remove the thing completely and make butler do evt.packet.body().content(); or shall we make it a method or shall we leave it as is?

Reply:
> shall we remove the thing completely and make butler do evt.packet.body().content(); or shall we make it a method or shall we leave it as is?

As a coder, do you know that `evt.packet.body().content()` off the bat is how to get a chat message's content? Personally, I would prefer to make it a method.

Reply:
ok sure